src/dll/edbgEcmdDll.C

```diff
@@ -1074,152 +970,50 @@ uint32_t dllCreateChipUnitScomAddress(const ecmdChipTarget & i_target, uint64_t
-    }
+  if (chipType == CHIP_TYPE_P9) {
+      rc = p9_dllQueryScom(i_target, o_queryData, i_address, i_detail);
+  } else {
+     return ECMD_FUNCTION_NOT_SUPPORTED;
   }
 
-  o_queryData.push_back(sdReturn);
+  if (rc) {
+      return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "queryscom failed!!");
+  }
 
   return rc;
 }
 
 uint32_t dllGetScom(const ecmdChipTarget & i_target, uint64_t i_address, ecmdDataBuffer & o_data) {
   uint32_t rc = ECMD_SUCCESS;
-  uint64_t data;
-  struct pdbg_target *target;
-
-  // Convert the input address to an absolute chip level address
-  i_address = getRawScomAddress(i_target, i_address);
-
-  // Now for the call to pdbg, use just the chip level target so the address
-  // doesn't get translated again down in pdbg
-  // i_target is pass by reference, make a local copy before we modify so we don't break upstream
-  ecmdChipTarget l_target = i_target;
-  l_target.chipUnitTypeState = ECMD_TARGET_FIELD_UNUSED;
-
-  // Get the chip level pdbg target for the call to the pib read
-  if (fetchPdbgTarget(l_target, &target)) {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unable to find PIB target\n");
-  }
 
-  // Make sure the pdbg target probe has been done and get the target state
-  if (pdbg_target_probe(target) != PDBG_TARGET_ENABLED) {
-    return out.error(ECMD_TARGET_NOT_CONFIGURED, FUNCNAME, "Target not configured!\n");
+  if (chipType == CHIP_TYPE_P9) {
+      rc = p9_dllGetScom(i_target,i_address,o_data);
+  } else {
+     return ECMD_FUNCTION_NOT_SUPPORTED;
   }
 
-  // Do the read and store the data in the return buffer
-  rc = pib_read(target, i_address, &data);
   if (rc) {
-    return out.error(EDBG_READ_ERROR, FUNCNAME, "pib_read of 0x%" PRIx64 " failed!\n", i_address);
+      return out.error(EDBG_READ_ERROR, FUNCNAME, "getscom failed!!");
   }
-  o_data.setBitLength(64);
-  o_data.setDoubleWord(0, data);
 
   return rc;
 }
 
 uint32_t dllPutScom(const ecmdChipTarget & i_target, uint64_t i_address, const ecmdDataBuffer & i_data) {
   uint32_t rc = ECMD_SUCCESS;
-  struct pdbg_target *target;
-
-  // Convert the input address to an absolute chip level address
-  i_address = getRawScomAddress(i_target, i_address);
-
-  // Now for the call to pdbg, use just the chip level target so the address
-  // doesn't get translated again down in pdbg
-  // i_target is pass by reference, make a local copy before we modify so we don't break upstream
-  ecmdChipTarget l_target = i_target;
-  l_target.chipUnitTypeState = ECMD_TARGET_FIELD_UNUSED;
 
-  // Get the chip level pdbg target for the call to the pib write
-  if (fetchPdbgTarget(l_target, &target)) {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unable to find PIB target\n");
-  }
-
-  // Make sure the pdbg target probe has been done and get the target state
-  if (pdbg_target_probe(target) != PDBG_TARGET_ENABLED) {
-    return out.error(ECMD_TARGET_NOT_CONFIGURED, FUNCNAME, "Target not configured!\n");
+  if (chipType == CHIP_TYPE_P9) {
```

Comment:
Is there a way to get from pdbg the type of chip it is working with?

Reply:
There is an API,  But all the changes are only pdbg downstream. since Alistair moved out of IBM not sure when it will get to upstream until unless someone manages upstream in place of Alistair

Reply:
This goes back to have a common function to get this information and have it return an actual enum value as well, instead of a string we are comparing against.